example_plugins/keyboard/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+use cimvr_common::{
+    desktop::{ElementState, InputEvent, InputEvents, KeyCode},
+    nalgebra::Vector3,
+    render::{Mesh, MeshHandle, Primitive, Render, UploadMesh, Vertex},
+    Transform,
+};
+use cimvr_engine_interface::{dbg, make_app_state, pkg_namespace, prelude::*, FrameTime};
+use serde::{Deserialize, Serialize};
+
+struct ServerState;
+
+struct ClientState {
+    w_is_pressed: bool,
```

Comment:
I'm confused why ClientState manages the `is_pressed` fields. Would it not make more sense to store this information with the MoveCommand? Or is ClientState being used instead of a local keypress struct? If the latter, would it be okay for the client state to not be used in place of a different struct for `is_pressed`? 

I know that this is most likely a common practice with ECS but what are the usecases for adding state/data to the ClientState struct?

Reply:
Only the client side knows how long the button is pressed for. Sending movement commands about what to do and for how long makes some sense here- we could include a 'time delta' field along with the buttons, but this was easier. 

We could absolutely make the button states their own struct too. In fact, having an abstraction which uses e.g. a hashmap internally to store which buttons are currently pressed would be nice. There's already an abstraction for winit we might make use of https://docs.rs/winit_input_helper/0.13.0/winit_input_helper/